Update replication docs to clarify permissions (#10536)
Minio-io/minio: docs/bucket/replication/README.md

Role ARN = 'arn:minio:replication:us-east-1:c5be6b16-769d-432a-9ef1-4567081f3566:destbucket'
```

> Note that the admin user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on source cluster if admin user is not the default minio admin. The access credentials used at the destination requires *s3:ReplicateObject* permission.

Reviewer: ```suggestion
> NOTE:  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster (we do not recommend running root credentials/super admin with replication). The access credentials used at the destination requires *s3:ReplicateObject* permission.
```

Minimal set of permissions needed on source bucket are
```
    "s3:GetReplicationConfiguration",
    "s3:ListBucket",
    "s3:GetBucketLocation",
    "s3:GetBucketVersioning"
```
The access key configured to perform replication on the target cluster should have following permissions on destination bucket.Note that the "s3:GetBucketVersioning" permission is specific to the destbucket resource, and remaining permissions pertain to resource prefix.
```
    "s3:ReplicateTags",
    "s3:GetObject",
    "s3:GetObjectVersion",
    "s3:GetObjectVersionTagging",
    "s3:PutObject",
    "s3:ReplicateObject"
    "s3:GetBucketVersioning"
```

> Once successfully created and authorized the `mc admin bucket remote add` command generates a replication target ARN.  The command below lists all the currently authorized replication targets:

Reviewer: It would be worth to provide an entire JSON document with full examples @poornas

Role ARN = 'arn:minio:replication:us-east-1:c5be6b16-769d-432a-9ef1-4567081f3566:destbucket'
```

> > NOTE:  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster (we do not recommend running root credentials/super admin with replication). The access credentials used at the destination requires *s3:ReplicateObject* permission.

Reviewer 2: ```suggestion
>  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster. We do not recommend running root credentials/super admin with replication, instead create a dedicated user. The access credentials used at the destination requires *s3:ReplicateObject* permission.
```

> Minimal policy on source bucket should have following permissions on the source bucket.

Reviewer 2: ```suggestion
The *source* bucket should have following minimal permission policy:
```

```
{
 "Version": "2012-10-17",
 "Statement": [
  {
   "Effect": "Allow",
   "Action": [
    "s3:GetReplicationConfiguration",
    "s3:ListBucket",
    "s3:GetBucketLocation",
    "s3:GetBucketVersioning"
   ],
   "Resource": [
    "arn:aws:s3:::srcbucket"
   ]
  }
}
```

> The access key configured to perform replication on the target cluster should have minimal permissions as in example below.

Reviewer 2: ```suggestion
The access key provided for the replication *target* cluster should have these minimal permissions:
```

```
{
 "Version": "2012-10-17",
 "Statement": [
  {
   "Effect": "Allow",
   "Action": [
    "s3:GetBucketVersioning"
   ],
   "Resource": [
    "arn:aws:s3:::destbucket"
   ]
  },
  {
   "Effect": "Allow",
   "Action": [
    "s3:ReplicateTags",
    "s3:GetObject",
    "s3:GetObjectVersion",
    "s3:GetObjectVersionTagging",
    "s3:PutObject",

>     "s3:ReplicateObject"

Reviewer 3: Do we need these permissions?

"s3:PutObjectRetention",
"s3:PutObjectLegalHold",
"s3:PutObjectTagging",

Author: only if object locking is enabled.

   ],
   "Resource": [
    "arn:aws:s3:::destbucket/*"
   ]
  }
 ]
}

```

> Once successfully created and authorized the `mc admin bucket remote add` command generates a replication target ARN.  The command below lists all the currently authorized replication targets:

Reviewer 2: ```suggestion
Once successfully created and authorized, the `mc admin bucket remote add` command generates a replication target ARN.  This command lists all the currently authorized replication targets:
```
